injector.py

```diff
@@ -0,0 +1,151 @@
+"""
+This Module is responsible for first downloading every file using the API Key , converting it into a string , then a file like object and processing it 
+the download function then injects the list of objects of the CovidHandler class into the covid database 
+"""
+
+import mysql.connector
+from mysql.connector import errorcode
+from Sqlqueries import SqlQueries
+import datetime
+from constants import API_URL
+import logging
+import requests
+from io import StringIO
+import csv
+from datetime import datetime
+from row_class import CovidRowHandler
+
+logger = logging.getLogger()
+
+logging.basicConfig(level=logging.INFO)
+
+
+class Inject:
+    """
+    The inject class takes in attributes for initialization cnx and cursor , both of which are required for communication with the database
+    They are not declared here but in the context manager to automatically enter and close the connection when required
+    """
+
+    cols_needed = [
+        "Province_State",
+        "Country_Region",
+        "Last_Update",
+        "Confirmed",
+        "Active",
+        "Deaths",
+        "Recovered",
+        "Incident_Rate",
+        "Case_Fatality_Ratio",
+    ]
+    columns_str = ", ".join(cols_needed)
+
+    def __init__(self, cnx, cursor):
+
+        self.cnx = cnx
+        self.cursor = cursor
+
+    def create_table(self):
+
+        Table = SqlQueries.Create_Table
+
+        try:
+            logger.info("Creating table")
+            self.cursor.execute(Table)
+            logger.info("Table created successfully")
+        except mysql.connector.Error as err:
+            if err.errno == errorcode.ER_TABLE_EXISTS_ERROR:
+                logger.error("the table already exists")
+            else:
+                logger.error("Error message: %s", err)
+
+    def rename_key(self, data_dict, old_key, new_key):
+        if old_key in data_dict:
+            data_dict[new_key] = data_dict[old_key]
+            del data_dict[old_key]
+
+    def download(self):
+        """
+        Each file is downloaded one by one .  A list of dictionaries (files) is iterated over to get the download URL for each file
+        Each file is then converted to a string using .text and then a stringio object . The reason for that is t oallow the file to be processed
+        without downloading it into the local disk .
+        Returns a list of objects , with each object being a row in a file . in total 3.5 million objects
+        the format is : [obj1 ,obj2 obj3]"
+        """
+        response = requests.get(API_URL)
+        files = response.json()
+        all_rows = []
+
+        for file_info in files:
+            file_rows = []
+
+            if file_info["type"] == "file":
+                file_name = file_info["name"]
+                print(f"The current file being processed is : {file_name}")
+                if ".csv" in file_name:
```

Comment:
Can we not achieve the same behavior with the following code?
```suggestion
            if not file_name.endswith(".csv"):
                continue
```